### PR TITLE
Docs: Drawer component

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds documentation and example for the Drawer component.

**Which issue(s) this PR fixes**:

Part of #25513

**Special notes for your reviewer**:
Would love feedback on how to expand the docs, struggled to come up with more information.

### CODE REVIEW

File: packages/grafana-ui/src/components/Drawer/Drawer.mdx [hunk 1]
```diff
@@ -1,4 +1,35 @@
-import { Props } from '@storybook/addon-docs/blocks';
+import { Meta, Props } from '@storybook/addon-docs/blocks';
 import { Drawer } from './Drawer';
 
+<Meta title="MDX|Drawer" component={Drawer} />
+
+# Drawer
+### When to use
+The Drawer component is a slide in overlay that can be used to display information without
+hiding content.
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Drawer/Drawer.mdx:9]:
```suggestion
Drawer is a slide in overlay that can be used to display additional information without hiding the main page content. It is usually anchored to the left or right edge of the screen.

```

File: packages/grafana-ui/src/components/Drawer/Drawer.mdx [hunk 2]
```diff
@@ -1,4 +1,35 @@
...
+
+One example of the Drawer in use is in Grafanas [panel inspector](https://grafana.com/docs/grafana/latest/panels/inspect-panel/).
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Drawer/Drawer.mdx:11]:
```suggestion
One example of the Drawer in use is in Grafana's [panel inspector](https://grafana.com/docs/grafana/latest/panels/inspect-panel/).
```

**Review by Clarity-89 — APPROVED** [review]
Looks good! Added some suggestions for the wording.

**Review by jessover9000 — COMMENTED** [review]
Cool! About the "When to use" part: Saying "It is usually anchored to the left or right edge of the screen" makes me wonder whether there's any other alternatives. Can the drawer component do top or bottom drawers/are there examples for that? (Like query history for example?) If not, we shouldn't say "is usually anchored" but should make the phrasing exclusive: "It can be anchored either to the left or right edge …". I would also add that the drawer cannot only display information but also contain interactive elements, or rather, it can contain pretty much anything you want right? 🤓 
And one thing not directly related to the docs: looking at the storybook canvas makes me wonder why that "open drawer" button is styled so weirdly. couldn't we put a proper Grafana button style there?

**Pull request merged**